src/app/helpers/attachment/attachmentLoader.ts

```diff
@@ -0,0 +1,134 @@
+import {
+    binaryStringToArray,
+    concatArrays,
+    decodeBase64,
+    decryptMessage,
+    decryptSessionKey,
+    getMessage
+} from 'pmcrypto';
+
+import { getAttachment } from '../../api/attachments';
+import { MessageExtended } from '../../models/message';
+import { Attachment, AttachmentData } from '../../models/attachment';
+import { AttachmentsCache } from '../../hooks/useAttachments';
+import { Api } from '../../models/utils';
+
+// Reference: Angular/src/app/attachments/services/AttachmentLoader.js
+
+interface SessionKey {
+    data?: Uint8Array;
+    algorithm?: string;
+}
+
+// TODO: Handle isOutside()
+
+export const decrypt = async (
+    encryptedBinaryBuffer: ArrayBuffer,
+    sessionKey: SessionKey = {}
+): Promise<AttachmentData> => {
+    const encryptedBinary = new Uint8Array(encryptedBinaryBuffer);
+
+    try {
+        return await decryptMessage({
+            message: await getMessage(encryptedBinary),
+            sessionKeys: [sessionKey],
+            format: 'binary'
+        });
+    } catch (err) {
+        console.error(err);
+        throw err;
+    }
+};
```

Comment:
Is this `try ... catch` needed? Wouldn't the code without it do the same?

Reply:
I think it will assure the error to be logged when it could be missed if it's in a promised with no catch.

But, to  be honest, the error management is quite a mess in general at this point. I need to get a special task for it if I don't really now how to organize that.